### ISSUE TITLE
Getting an unexpected error from torrent-add?

### ISSUE THREAD

**Issue by jethrosun** [j0rsa/transmission-rpc#4]
Hi, 

Sometime when I try to add a torrent I will get an error like below:
```
cargo r --example torrent-add
   Compiling transmission-rpc v0.2.3 (/home/jethros/dev/pvn/transmission-rpc)
    Finished dev [unoptimized + debuginfo] target(s) in 1.93s
     Running `target/debug/examples/torrent-add`
Error: reqwest::Error { kind: Decode, source: Error("missing field `torrent-added`", line: 1, column: 141) }
```

I guess this is not expected since in the example code this Error is not being captured? Note that sometime I can get the corrent response but most of the times it is this error. Trying to run torrent-remove prior is not helping either. 

Thx

**Comment by red-avtovo** [comment]
Hi @jethrosun 
It seems, that the library receives a completely unexpected response which doesn't comply with the spec. I believe there should be something like a network misconfiguration. Can you try to send the same request using for example Postman and see what is the error response you receiving?

**Comment by jethrosun** [comment]
Yeah, I can dig deeper with postman in a bit. Other examples actually work fine so I don't whether it is actually a network misconfig

**Comment by red-avtovo** [comment]
I ran some tests and found out that when you have malformed URL or incorrect data, `torrent-add` method panics since it doesn't receive back any arguments in the response. 
Thank you for your issue, it is fixed in version [`0.3.0`](https://crates.io/crates/transmission-rpc)

**Comment by jethrosun** [comment]
Thank you :)

I think we can close it then
